http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoMetricsMonitor.scala

```diff
@@ -235,22 +256,143 @@ class LeoMetricsMonitor[F[_]](config: LeoMetricsMonitorConfig,
+        .flatMap(_.apps)
+        .exists(a => a.status != AppStatus.Deleted)
+    } yield List(c)
+
+    activeClusters.combineAll
+      .parTraverseN(parallelism) { case cluster =>
+        for {
+          ctx <- ev.ask
+          azureCloudContext <- F.fromOption(
+            cloudContextToManagedResourceGroup.get(cluster.cloudContext),
+            new RuntimeException(s"Azure cloud context not found for cluster ${cluster}: ${ctx.traceId}")
+          )
+          clusterName = AKSClusterName(cluster.clusterName.value)
+          cluster <- azureContainerService.getCluster(clusterName, azureCloudContext).attempt
+          res <- cluster match {
+            case Left(_) =>
+              logger
+                .warn(ctx.loggingCtx)(
+                  s"Cluster ${azureCloudContext.asString} / ${clusterName} does not exist. Skipping metrics collection."
+                )
+                .as(List.empty[Map[NodepoolSizeMetric, Double]])
+            case Right(c) =>
+              F.delay(c.agentPools().asScala.toList.map { case (name, pool) =>
+                Map(NodepoolSizeMetric(azureCloudContext, name) -> pool.count.doubleValue)
+              })
+          }
+        } yield res.combineAll
+      }
+      .map(_.combineAll)
+  }
+
+  /**
+   * Records memory/cpu requests/limits by (cloud, appType, service).
+   * Only Azure apps supported.
+   */
+  private[monitor] def getAppK8sResources(allClusters: List[KubernetesCluster])(implicit
+    ev: Ask[F, AppContext]
+  ): F[Map[AppResourcesMetric, Double]] = {
+    val allServices = for {
+      // TODO: handle GCP
+      c <- allClusters if c.cloudContext.cloudProvider == CloudProvider.Azure
+      n <- c.nodepools
+      // Only care about Running apps for resource metrics
+      a <- n.apps if a.status == AppStatus.Running
+    } yield Map((c.clusterName, c.cloudContext) -> List(a))
+
+    allServices.combineAll.toList
+      .parTraverseN(parallelism) { case ((clusterName, cloudContext), apps) =>
+        for {
+          ctx <- ev.ask
+
+          // Build k8s client
+          azureCloudContext <- F.fromOption(
+            cloudContextToManagedResourceGroup.get(cloudContext),
+            new RuntimeException(s"Azure cloud context not found for cluster ${clusterName}: ${ctx.traceId}")
+          )
+          aksClusterName = AKSClusterName(clusterName.value)
+          client <- kubeAlg.createAzureClient(azureCloudContext, aksClusterName).attempt
+
+          res <- client match {
+            case Left(_) =>
+              logger
+                .warn(ctx.loggingCtx)(
+                  s"Cluster ${azureCloudContext.asString} / ${clusterName} does not exist. Skipping metrics collection."
+                )
+                .as(List.empty[Map[AppResourcesMetric, Double]])
+            case Right(client) =>
+              // For each app, query pods by leoAppName label and services by leoServiceName label.
```

Comment:
See PRs for setting the `leoServiceName` label:
https://github.com/broadinstitute/terra-helmfile/pull/4288
https://github.com/broadinstitute/cromwhelm/pull/264
https://github.com/DataBiosphere/terra-app/pull/38/